File: src/PalindromeNumber.java
// https://leetcode.com/problems/palindrome-number/

import java.util.Scanner;

public class PalindromeNumber {

    public boolean isPalindrome(int number) {
        if (number < 0 || (number % 10 == 0 && number != 0)) {
            return false;
        }
        int reverse = 0;
        while (number > reverse) {
            reverse = 10 * reverse + number % 10;
            number /= 10;
        }
        return number == reverse || number == reverse / 10;
    }
}

File: javascript/PalindromeNumber.js
// https://leetcode.com/problems/palindrome-number
// T: O(log(n))
// S: O(1)

const isPalindrome = x => {
  if (x < 0 || (x % 10 === 0 && x !== 0)) {
    return false;
  }
  let result = x;
  let sum = 0;
  while (x !== 0) {
    let remainder = x % 10;
    sum = sum * 10 + remainder;
    x = Math.floor(x / 10);
  }

  return sum === result;
};
